### ISSUE TITLE
Could not initialize class org. Owasp. Esapi. Reference. DefaultValidator

### ISSUE THREAD

**Issue by SuperM-L** [ESAPI/esapi-java-legacy#760]
Hi, I used: "ESAPI.validator()" to get the "Could not initialize class org. Owasp. Esapi. Reference. DefaultValidator" exception, from the version 2.1.0 upgraded to 2.5.0.0, I watched 2.2.1.0 release notes and configured: "org.owasp.esapi.logging.java.JavaLogFactory - To use the new default, java.util.logging (JUL)" 
and 
"
Logger.ApplicationName=ExampleApplication
Logger.LogEncodingRequired=false
Logger.LogApplicationName=true
Logger.LogServerIP=true
Logger.LogFileName=ESAPI_logging_file
Logger.MaxLogFileSize=10000000
Logger.UserInfo=true
Logger.ClientInfo=true
"

But the problem still arises. Does anyone know how to solve it

**Comment by jayapal-ganupu** [comment]
i am using esapi-2.5.0.0. 
facing below issue . please help on this
02-04T09:42:02.263451085Z: [ERROR]  Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.owasp.esapi.reference.DefaultValidator
2023-02-04T09:42:02.264467484Z: [ERROR]      at java.base/java.lang.Class.forName0(Native Method)
2023-02-04T09:42:02.265341883Z: [ERROR]      at java.base/java.lang.Class.forName(Class.java:315)
2023-02-04T09:42:02.265360783Z: [ERROR]      at org.owasp.esapi.util.ObjFactory.loadClassByStringName(ObjFactory.java:158)
2023-02-04T09:42:02.269216579Z: [ERROR]      at org.owasp.esapi.util.ObjFactory.make(ObjFactory.java:81)
2023-02-04T09:42:02.269235679Z: [ERROR]      at org.owasp.esapi.ESAPI.validator(ESAPI.java:192)

**Comment by vincent-simpson** [comment]
@SuperM-L how did you resolve this?

**Comment by andyyyyy** [comment]
using 2.5.0.2 , have exact same issue, anyone got a solution?
I check the org.owasp.esapi.reference.DefaultValidator do exist, but it throws ClassNotFoundException

**Comment by kwwall** [comment]
@SuperM-L, @vincent-simpson, @andyyyyy - I am not sure why @SuperM-L closed this issue, but it he hadn't, then one of the ESAPI dev team members likely would have. We will answer questions on our mailing lists or in the Q&A Discussions forum, but not in the issues. (We sometimes will answer when posted to StackOverflow as well, but no promises for that forum.) However, opening issues to questions or general pleas of assistance makes it ugly for writing them up in the release notes so if we see that, we will close the issue and re-post it to the Discussion board.

That said, if any of you wish to turn this into a legitimate GitHub issue, we are going to need you to submit it to us in a manner that we can reproduce, which means a code example, directions of how you are compiling it and running it (or provided scripts), etc. That said, either post this to the Discussion (and even there, we at least still need a LOT more info--e.g., a FULL stack exception as usually there are multiple "Caused by..." that you are not showing but point to the root cause).

**Comment by SuperM-L** [comment]
> 
@kwwall @andyyyyy @vincent-simpson @jayapal-ganupu 
I’m sorry for just seeing your message now. The reason for closing this topic is that I had already resolved it in December last year. Let me think about how I resolved it. I will provide you with my code changes, which I remember were made based on the official update instructions. I have forgotten the specific details, so please forgive me.I hope this is helpful to you.

- First, there is this ESAPI.properties file, and I made modifications in the following two places.
1. first:Modify
![1](https://github.com/ESAPI/esapi-java-legacy/assets/47932027/c10122a7-fbed-4da8-821d-94cc069a35e0)
code:`
SAPI.Logger=org.owasp.esapi.logging.java.JavaLogFactory
`
2. second :Add
![2](https://github.com/ESAPI/esapi-java-legacy/assets/47932027/9cf52a1d-ed9c-4ffc-98be-706eb71a0067)
code:
`
Logger.UserInfo=true
`
`
Logger.ClientInfo=true
`
- Then there is this esapi-java-logging.properties file, which may not have existed before the upgrade. Now, in order to address the issue I raised, it is necessary to create this esapi-java-logging.properties file in the same directory as the ESAPI.properties file.
![3](https://github.com/ESAPI/esapi-java-legacy/assets/47932027/10cf3c98-7cf7-46be-9214-329d068577cd)
code:
`
handlers=java.util.logging.ConsoleHandler
`
`
.level=INFO
`
`
java.util.logging.ConsoleHandler.level=INFO
`
`
java.util.logging.ConsoleHandler.formatter=java.util.logging.SimpleFormatter
`
`
java.util.logging.SimpleFormatter.format=[%1$tF %1$tT] [%3$-7s] %5$s %n
`

**Comment by kwwall** [comment]
Okay, well thanks for wrapping that up and letting us know. That's what I expected it was all along but details were missing. Had we seen the _full_ stack exception trace, there would have been a exception that would have pointed to `org.owasp.esapi.reference.JavaLogFactory` as the root cause, and had I noticed that, I just would have referred you to https://github.com/ESAPI/esapi-java-legacy/wiki/Configuration-Reference:-JavaLogFactory on our GitHub wiki. That's quite a common mistake and @xeno6696 and I have commented on it many times on Stack Overflow.
